### PR TITLE
udp-notifier feature

### PR DESCRIPTION
**Pull request by islamoglus**
Hi,
I need to get information about failed turn command request for build some statistics.

This implementation provides a generic notification interface for Turn server.
UDP interface is currently implemented, but different notification interfaces, such as SNMP, can be applied in the future.

Notification interface is used for "executed turn command" currently, but it can be used for different type of notification messages in the future.

Thanks for reviewing.
Selahaddin.

### CODE REVIEW

**Comment by misi** [comment]
@islamoglus I am traveling this week, I will to check it ASAP..
Thanks for you patience..

File: src/apps/relay/ntfy_drivers/udp_subscriber.c [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+    if ( data->hostent == NULL) { 
+        return NULL;
+    }
+    
+    data->sockfd = socket(AF_INET, SOCK_DGRAM, 0);
+    if (data->sockfd == -1) {
+        return NULL;
+    }
+
+    data->server_addr.sin_family = AF_INET;
```
Comment (misi) [inline src/apps/relay/ntfy_drivers/udp_subscriber.c]: What happens if somebody want to log to IPv6 address? 
e.g. AF_INET6. May not good idea to "hardcode" Address Family.

**Comment by misi** [comment]
Some documentation needed in README.turnserver file.
I need more time to understand every details in your code.

File: src/apps/relay/ntfy_drivers/udp_subscriber.c (hunk 1 above)
Comment (islamoglus) [reply src/apps/relay/ntfy_drivers/udp_subscriber.c]: resolved

**Comment by islamoglus** [comment]
Hi @misi , is there any update about this pull request. Is there anything do you want that needed for merge.

**Comment by misi** [comment]
@islamoglus  Sorry. I only need time to review. 
Keep pinging me and I will do it. :) 
Thanks..

**Comment by bradleythughes** [comment]
I don't see any explanation/documentation of the message format, or any examples of what people can do with this feature. This makes me very reluctant to add it as-is.

If you are looking to build or gather statistics, why not add the statistic you are interested in to the rest of the ones put into Redis?

**Comment by islamoglus** [comment]
I will add detailed descriptions ASAP.
Thanks for review

**Comment by Neustradamus** [comment]
@islamoglus: There is a conflict :/

@misi: Have you seen?

**Comment by islamoglus** [comment]
closed

**Comment by Neustradamus** [comment]
@islamoglus: Why have you closed this ticket?

**Comment by misi** [comment]
@islamoglus You are not interested in this feature, or you get rid that it is not yet mereged?
I will try to dry the PR queue this week..

**Comment by misi** [comment]
@islamoglus Thanks! I will close it then and let me know if you want to reopen it..

**Pull request closed without merging**